packages/core/template/template.js

```diff
@@ -5,46 +5,50 @@ const fsUtil = require('../src/utils/fsUtil');
 
 const requiredFiles = ['index.md', 'site.json', '_markbind/'];
 
-function Template(rootPath, templatePath) {
-  this.rootPath = rootPath;
-  this.template = templatePath;
-}
+class Template {
+  constructor(rootPath, templatePath) {
+    this.root = rootPath;
+    this.template = templatePath;
+  }
 
-function validateTemplateFromPath(templatePath) {
-  for (let i = 0; i < requiredFiles.length; i += 1) {
-    const requiredFile = requiredFiles[i];
-    const requiredFilePath = path.join(templatePath, requiredFile);
+  static validateTemplateFromPath(templatePath) {
+    for (let i = 0; i < requiredFiles.length; i += 1) {
+      const requiredFile = requiredFiles[i];
+      const requiredFilePath = path.join(templatePath, requiredFile);
 
-    if (!fs.existsSync(requiredFilePath)) {
-      return false;
+      if (!fs.existsSync(requiredFilePath)) {
+        return false;
+      }
     }
+
+    return true;
   }
 
-  return true;
-}
+  static generateSiteWithTemplate(rootPath, templatePath) {
+    return new Promise((resolve, reject) => {
+      fs.access(rootPath)
+        .catch(() => fs.mkdirSync(rootPath))
+        .then(() => fsUtil.copySyncWithOptions(templatePath, rootPath, { overwrite: false }))
+        .then(resolve)
+        .catch(reject);
+    });
+  }
 
-function generateSiteWithTemplate(rootPath, templatePath) {
-  return new Promise((resolve, reject) => {
-    fs.access(rootPath)
-      .catch(() => fs.mkdirSync(rootPath))
-      .then(() => fsUtil.copySyncWithOptions(templatePath, rootPath, { overwrite: false }))
-      .then(resolve)
-      .catch(reject);
-  });
-}
+  init() {
+    const templatePath = path.join(__dirname, this.template);
```

Comment:
How about moving this up to constructor and refactor accordingly?

Reply:
Good suggestion - thank you!

Reply:
Updated in [38c7676](https://github.com/MarkBind/markbind/pull/2148/commits/38c76764fed6da189b9c3cc5a76ef64d8f2a40b0)!